server/embed/config.go

```diff
@@ -897,6 +916,36 @@ func (cfg *configYAML) configFromFile(path string) error {
 	return cfg.Validate()
 }
 
+// SetFeatureGatesFromExperimentalFlags sets the feature gate values if the feature gate is not explicitly set
+// while their corresponding experimental flags are explicitly set, for all the features in ExperimentalFlagToFeatureMap.
+// TODO: remove after all experimental flags are deprecated.
+func SetFeatureGatesFromExperimentalFlags(fg featuregate.FeatureGate, getExperimentalFlagVal func(string) *bool, featureGatesFlagName, featureGatesVal string) error {
```

Comment:
You should be able to read `embed.ServerFeatureGateFlagName` directly.
```suggestion
func SetFeatureGatesFromExperimentalFlags(fg featuregate.FeatureGate, getExperimentalFlagVal func(string) *bool, featureGatesVal string) error {
```

Reply:
this is passed as a param because we might want to reuse this function for `cluster-feature-gates`.

Reply:
We haven't finalized the cluster-level feature yet. We can make change later once it's finalized.

Reply:
ok, removed.